packages/core/src/amazonqGumby/chat/controller/messenger/messenger.ts

```diff
@@ -0,0 +1,358 @@
+        if (credentialState.amazonQ === 'expired') {
+            authType = 're-auth'
+            message = expiredText
+        }
+
+        this.dispatcher.sendAuthNeededExceptionMessage(new AuthNeededException(message, authType, tabID))
+    }
+
+    public sendAuthenticationUpdate(gumbyEnabled: boolean, authenticatingTabIDs: string[]) {
+        this.dispatcher.sendAuthenticationUpdate(new AuthenticationUpdateMessage(gumbyEnabled, authenticatingTabIDs))
+    }
+
+    public async sendProjectPrompt(modules: TransformationCandidateProject[], tabID: string) {
+        const moduleFormOptions: { value: string; label: string }[] = []
+        const uniqueJavaOptions = new Set<JDKVersion>()
+
+        modules.forEach(candidateModule => {
+            moduleFormOptions.push({
+                value: candidateModule.path,
+                label: candidateModule.name,
+            })
+
+            if (candidateModule.JDKVersion !== undefined) {
+                uniqueJavaOptions.add(candidateModule.JDKVersion)
+            }
+        })
+
+        const jdkFormOptions: { value: string; label: string }[] = []
+        uniqueJavaOptions.forEach(jdkVersion => {
+            jdkFormOptions.push({
+                value: jdkVersion.toString(),
+                label: jdkVersion.toString(),
+            })
+        })
+
+        const formItems: ChatItemFormItem[] = []
+        formItems.push({
+            id: 'GumbyTransformModuleForm',
+            type: 'select',
+            title: 'Choose a module to transform',
+            mandatory: true,
+            options: moduleFormOptions,
+        })
+
+        formItems.push({
+            id: 'GumbyTransformJdkFromForm',
+            type: 'select',
+            title: 'Choose the source code version',
+            mandatory: true,
+            options: jdkFormOptions,
+        })
+
+        formItems.push({
+            id: 'GumbyTransformJdkToForm',
+            type: 'select',
+            title: 'Choose the target code version',
+            mandatory: true,
+            options: [
+                {
+                    value: JDKVersion.JDK17.toString(),
+                    label: JDKVersion.JDK17.toString(),
+                },
+            ],
+        })
+
+        this.dispatcher.sendAsyncEventProgress(
+            new AsyncEventProgressMessage(tabID, {
+                inProgress: true,
+                message: `I can upgrade your Java ${jdkFormOptions[0].label} project. To start the transformation, I need some information from you. Choose the module you want to upgrade and the target code version to upgrade to, and then choose Transform. It can take 10-30 minutes to upgrade your code, depending on the size of your module.`,
+            })
+        )
+
+        this.dispatcher.sendChatPrompt(
+            new ChatPrompt(
+                {
+                    message: 'Q Code Transformation',
+                    formItems: formItems,
+                },
+                `TransformForm`,
```

Comment:
Nit: No need for backtick to denote a string literal, could just be normal string syntax `"Transform"` or `'Transform'`